### PR TITLE
Base32 decoding function updated.

### PR DESCRIPTION
**Pull request by IrinaRakh**
Return nil if a symbol is not from the table instead of skipping the symbol.

### CODE REVIEW

File: YubiKit/YubiKit/ThirdParties/MF_Base32Additions.m [hunk 1]
```diff
@@ -62,7 +62,7 @@ +(NSData *)dataFromBase32String:(NSString *)encoding
                 if( c == '=' ) break; // padding...
 
                 c = decodingTable[c];
-                if( c == __ ) continue;
```
Comment (imakhalova) [inline YubiKit/YubiKit/ThirdParties/MF_Base32Additions.m]: we should not modify 3rd party lib...let's do the check on side of YubiKit which uses this method

File: YubiKit/YubiKit/Helpers/Additions/YKFNSDataAdditions.m [hunk 2]
```diff
@@ -248,7 +248,14 @@ - (BOOL)ykf_containsRange:(NSRange) range {
 @implementation NSData(NSData_Base32Additions)
 
 + (NSData *)ykf_dataWithBase32String:(NSString *)base32String {
-    return [self dataWithBase32String:base32String];
+    NSRange range = NSMakeRange(0, [base32String length]);
+    NSError *error = NULL;
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern: @"[A-Za-z2-7=]*" options: 0 error: &error];
+    NSRange foundRange = [regex rangeOfFirstMatchInString: base32String options: 0 range: range];
```
Comment (imakhalova) [inline YubiKit/YubiKit/Helpers/Additions/YKFNSDataAdditions.m]: this seems to return your first match...which will be your first character, no?

Comment (imakhalova) [reply YubiKit/YubiKit/Helpers/Additions/YKFNSDataAdditions.m]: I think you should check the opposite...if range is not found then return nil
Otherwise you method might eturns nil for correct encoded string

File: YubiKit/YubiKit/Helpers/Additions/YKFNSDataAdditions.m [hunk 3]
```diff
@@ -248,7 +248,14 @@ - (BOOL)ykf_containsRange:(NSRange) range {
 @implementation NSData(NSData_Base32Additions)
 
 + (NSData *)ykf_dataWithBase32String:(NSString *)base32String {
```
Comment (imakhalova) [inline YubiKit/YubiKit/Helpers/Additions/YKFNSDataAdditions.m]: Can you please add couple UTs for this method...which will should that correct string works and incorrect one returns nil

File: YubiKit/YubiKit/Helpers/Additions/YKFNSDataAdditions.m (hunk 2 above)
Comment (IrinaRakh) [reply YubiKit/YubiKit/Helpers/Additions/YKFNSDataAdditions.m]: it returns a range of first continuous match

File: YubiKit/YubiKitTests/Tests/YKFNSDataAdditionsTests.m [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+- (void)test_WhenSecretContainsNumbersNotFromTheTable_ObjectNil {
+    NSString *secret = @"0189";
+    NSData *result = [NSData ykf_dataWithBase32String: secret];
+    
+    XCTAssertNil(result, @"Returned nil because the secret could not be parsed");
+}
+
+- (void)test_WhenSecretContainsPaddingAndSymbolsFromTheTable_ObjectNotNil {
+    NSString *secret = @"a=2";
```
Comment (imakhalova) [inline YubiKit/YubiKitTests/Tests/YKFNSDataAdditionsTests.m]: this is actually not invalid secret....encoded string can be finished with = signs...
But it's ok if this will still work (very edge case), but it's better to provide correct format of string for the test

File: YubiKit/YubiKitTests/Tests/YKFNSDataAdditionsTests.m [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+- (void)test_WhenSecretContainsNumbersNotFromTheTable_ObjectNil {
+    NSString *secret = @"0189";
+    NSData *result = [NSData ykf_dataWithBase32String: secret];
+    
+    XCTAssertNil(result, @"Returned nil because the secret could not be parsed");
+}
+
+- (void)test_WhenSecretContainsPaddingAndSymbolsFromTheTable_ObjectNotNil {
```
Comment (imakhalova) [inline YubiKit/YubiKitTests/Tests/YKFNSDataAdditionsTests.m]: nit: Can you keep successful scenarios grouped before failure cases?

File: YubiKit/YubiKitTests/Tests/YKFNSDataAdditionsTests.m [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    NSData *result = [NSData ykf_dataWithBase32String: secret];
+    
+    XCTAssertNotNil(result, @"Returned data object from a Base32 encoded string");
+}
+
+- (void)test_WhenSecretContainsSomeSymbolsNotFromTheTableAndSomeNot_ObjectNil {
+    NSString *secret = @"AAA111";
+    NSData *result = [NSData ykf_dataWithBase32String: secret];
+    
+    XCTAssertNil(result, @"Returned nil because the secret could not be parsed");
```
Comment (imakhalova) [inline YubiKit/YubiKitTests/Tests/YKFNSDataAdditionsTests.m]: because contains symbols that could not be decoded
